src/index.js

```diff
@@ -16,8 +16,14 @@ function init( {
 			createPopup( popupContainer ),
 		events = customEvents( popup ),
 		showPopup = ( { target } ) => {
+			if ( popup.timeoutId ) {
+				clearTimeout( popup.timeoutId )
+				popup.timeoutId = null
+			}
```

Comment:
it works basically but it doesn't clear the event listener through `hidePopup`. 

We can check whether the popup exists, then call the event `hidePopup` before showing the new popup, in this case we don't need to expose `timeoutId` as the attribute of popup.

Reply:
still not ready to test, right?
when it is please move it to QA and lmk?